src/libsyntax/mut_visit.rs

```diff
@@ -300,6 +284,42 @@ pub trait MutVisitor: Sized {
     fn visit_span(&mut self, _sp: &mut Span) {
         // Do nothing.
     }
+
+    fn flat_map_arg(&mut self, arg: Arg) -> SmallVec<[Arg; 1]> {
+        noop_flat_map_arg(arg, self)
+    }
+
+    fn flat_map_arm(&mut self, arm: Arm) -> SmallVec<[Arm; 1]> {
+        noop_flat_map_arm(arm, self)
+    }
+
+    fn flat_map_field(&mut self, f: Field) -> SmallVec<[Field; 1]> {
+        noop_flat_map_field(f, self)
+    }
+
+    fn flat_map_field_pattern(&mut self, fp: FieldPat) -> SmallVec<[FieldPat; 1]> {
+        noop_flat_map_field_pattern(fp, self)
+    }
+}
+
+pub fn noop_flat_map_field_pattern<T: MutVisitor>(
```

Comment:
This file uses some grouping for functions, e.g. all the `fn flat_map_*` functions are grouped together, all the `fn noop_flat_map_*` are grouped together as well.
Could you move the new code around a bit, so that the grouping is kept?

Reply:
Actually, disregard that, the diff will be entirely unreadable, and grouping by the "one vs many" property is not very useful.

It's better to keep all the *changed* functions in their place and apply the grouping only to the *new* ones.